src/Compression/Inflater.cs

```diff
@@ -0,0 +1,223 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security;
+
+using ObjCRuntime;
+
+namespace Compression
+{
+	/// <summary>
+	/// Provides a wrapper around the ZLib decompression API.
+	/// </summary>
+	internal sealed class Inflater : IDisposable
+	{
+
+		[StructLayout(LayoutKind.Sequential)]
+		struct CompressionStreamStruct {
+			public IntPtr Destination; // uint8_t * dst_ptr
+			public nint DestinationSize; // size_t dst_size;
+			public IntPtr Source; // const uint8_t * src_ptr;
+			public nint SourceSize; // size_t src_size;
+			public IntPtr State; // void * __nullable state;
+		}
+
+		[DllImport (Constants.libcompression)]
+		static extern CompressionStatus compression_stream_init (ref CompressionStreamStruct stream, StreamOperation operation, CompressionAlgorithm algorithm);
+
+		[DllImport (Constants.libcompression)]
+		static extern CompressionStatus compression_stream_process (ref CompressionStreamStruct stream, StreamFlag flags);
+
+		[DllImport (Constants.libcompression)]
+		static extern CompressionStatus compression_stream_destroy (ref CompressionStreamStruct stream);
```

Comment:
don't duplicate the p/invokes

Reply:
Yeah, it was in my todo after I got the tests to pass :)